sinks/influxdb/driver.go

```diff
@@ -37,72 +37,130 @@ type influxdbSink struct {
 	seqNum        metricSequenceNum
 }
 
-func (self *influxdbSink) Register(metrics []sink_api.MetricDescriptor) error {
+func (sink *influxdbSink) Register(metrics []sink_api.MetricDescriptor) error {
 	// Create tags once influxDB v0.9.0 is released.
 	return nil
 }
 
-func (self *influxdbSink) metricToSeries(timeseries *sink_api.Timeseries) *influxdb.Series {
-	columns := []string{}
-	values := []interface{}{}
-	// TODO: move labels to tags once v0.9.0 is released.
-	seriesName := timeseries.Point.Name
-	if timeseries.MetricDescriptor.Units.String() != "" {
-		seriesName = fmt.Sprintf("%s_%s", seriesName, timeseries.MetricDescriptor.Units.String())
-	}
-	if timeseries.MetricDescriptor.Type.String() != "" {
-		seriesName = fmt.Sprintf("%s_%s", seriesName, timeseries.MetricDescriptor.Type.String())
-	}
+// Returns true if this sink supports metrics
+func (sink *influxdbSink) SupportsMetrics() bool {
+	return true
+}
 
-	// Add the real metric value.
-	columns = append(columns, "value")
-	values = append(values, timeseries.Point.Value)
-	// Append labels.
-	if !self.avoidColumns {
-		for key, value := range timeseries.Point.Labels {
-			columns = append(columns, key)
-			values = append(values, value)
+// Returns true if this sink supports events
+func (sink *influxdbSink) SupportsEvents() bool {
+	return true
+}
+
+func (sink *influxdbSink) StoreTimeseries(timeseriesLists []*[]sink_api.Timeseries) error {
+	precisions := []sink_api.TimePrecision{sink_api.Second, sink_api.Millisecond, sink_api.Microsecond}
+	for _, precision := range precisions {
+		err := sink.storeTimeseriesWithPrecision(timeseriesLists, precision)
+		if err != nil {
+			return err
 		}
-	} else {
-		seriesName = strings.Replace(seriesName, "/", "_", -1)
-		seriesName = fmt.Sprintf("%s_%s", sink_api.LabelsToString(timeseries.Point.Labels, "_"), seriesName)
 	}
-	// Add timestamp.
-	columns = append(columns, "time")
-	values = append(values, timeseries.Point.End.Unix())
-	// Ass sequence number
-	columns = append(columns, "sequence_number")
-	values = append(values, self.seqNum.Get(seriesName))
-
-	return self.newSeries(seriesName, columns, values)
+	return nil
 }
 
-func (self *influxdbSink) StoreTimeseries(timeseries []sink_api.Timeseries) error {
+func (sink *influxdbSink) storeTimeseriesWithPrecision(timeseriesLists []*[]sink_api.Timeseries, precision sink_api.TimePrecision) error {
 	dataPoints := []*influxdb.Series{}
-	for index := range timeseries {
-		dataPoints = append(dataPoints, self.metricToSeries(&timeseries[index]))
+	for _, timeseriesList := range timeseriesLists {
+		if timeseriesList != nil {
+			for _, timeseries := range *timeseriesList {
+				if timeseries.TimePrecision == precision {
```

Comment:
Why do we need this check? Why can't we use the precision that is passed via `timeseries.TimePrecision`?